### PR TITLE
Add caching

### PR DESCRIPTION
**Pull request by bashtian**

### CODE REVIEW

File: graph/gaedatastore/quadstore.go [hunk 1]
```diff
@@ -461,9 +521,14 @@ func (qs *QuadStore) Size() int64 {
 		glog.Error("Error fetching size, context is nil, graph not correctly initialised")
 		return 0
 	}
+	//qs.context.Infof("Size %v", qs)
```
Comment (panamafrancis) [inline graph/gaedatastore/quadstore.go]: errant debug message

File: graph/gaedatastore/quadstore.go [hunk 2]
```diff
@@ -89,6 +114,35 @@ func init() {
...
+	cache = make(map[string]Cache)
+	cacheTime = time.Now()
+	t, _ := cacheTime.MarshalBinary()
+	item := &memcache.Item{
+		Key:   "last_cache_reset",
+		Value: t,
+	}
+	// Add the item to the memcache, if the key does not already exist
+	if err := memcache.Add(qs.db.Context, item); err == memcache.ErrNotStored {
+		qs.db.Context.Infof("item with key %q already exists", item.Key)
```
Comment (panamafrancis) [inline graph/gaedatastore/quadstore.go]: This logs during normal operation, perhaps remove it.

File: graph/gaedatastore/quadstore.go [hunk 3]
```diff
@@ -47,13 +51,28 @@ var (
...
+	cacheTime = time.Now()
 )
 
 type QuadStore struct {
-	context appengine.Context
+	hashSize   int
+	makeHasher func() hash.Hash
+	context    appengine.Context
+	db         *goon.Goon
+	//cache      map[string]Cache
```
Comment (panamafrancis) [inline graph/gaedatastore/quadstore.go]: remove the commented out code

**Comment by panamafrancis** [comment]
Is there any reasoning behind the 'cache' map[string]cache{} being implemented? I see it in the quadstore and iterator but it doesn't do anything. Also the checkCache() isn't called anywhere and from following the logic I noticed it will always fire resetCache()  so that should be removed. The resetCache() method should probably be removed too, as to my knowledge items fall out of memcache on a granular basis so this mechanism won't provide the needed behaviour. So along with that should go cacheTime, the cache struct {} and the inclusion of 'appengine/memcache' isn't needed.

Otherwise it looks ok!

**Comment by panamafrancis** [comment]
Also, could you add "github.com/mjibson/goon" to the .travis.yml

**Comment by panamafrancis** [comment]
Any change here? :)

**Comment by bashtian** [comment]
Was still on my todo list, but the list is to long ;) My idea for checkCache was to reset the cache when other instances modify the graph, but I'm not sure if this is the best solution. I removed it for now, it need more time for consideration.

**Comment by bashtian** [comment]
Moved to https://github.com/google/cayley/pull/334

**Pull request closed without merging**